arches/app/models/graph.py

```diff
@@ -496,42 +496,63 @@ def add_function(self, function):
         return function
 
     def add_resource_instance_lifecycle(self, resource_instance_lifecycle):
-        self.resource_instance_lifecycle = models.ResourceInstanceLifecycle(
-            id=resource_instance_lifecycle["id"],
-            name=resource_instance_lifecycle["name"],
+        """
+        Adds a ResourceInstanceLifecycle to this graph
+
+        Arguments:
+        resource_instance_lifecycle -- a dictionary representing a models.ResourceInstanceLifecycle instance
+
+        """
+
+        resource_instance_lifecycle_query = (
+            models.ResourceInstanceLifecycle.objects.filter(
+                pk=resource_instance_lifecycle["id"]
+            )
         )
 
-        resource_instance_lifecycle_states = []
-        for resource_instance_lifecycle_state_json in resource_instance_lifecycle[
-            "resource_instance_lifecycle_states"
-        ]:
-            next_resource_instance_lifecycle_states = (
-                resource_instance_lifecycle_state_json.pop(
-                    "next_resource_instance_lifecycle_states"
-                )
+        if resource_instance_lifecycle_query.exists():
+            self.resource_instance_lifecycle = resource_instance_lifecycle_query.first()
```

Comment:
exists() just adds a query if you end up evaluating anyway, so I'd just take first() and compare to None.